deepset_cloud_sdk/_service/files_service.py

```diff
@@ -363,45 +342,26 @@ def _remove_duplicates(file_paths: List[Path]) -> List[Path]:
 
         return most_recent_files
 
-    @staticmethod
-    def _get_allowed_file_types(desired_file_types: Optional[List[Any]]) -> List[str]:
-        """Filter `SUPPORTED_TYPE_SUFFIXES` by `desired_file_types`.
-
-        If desired_file_types is empty, all supported file types are returned.
-        :param desired_file_types: A list of desired file types.
-        :return: A list of desired file types that can be processed by deepset Cloud.
-        """
-        if not desired_file_types:
-            return SUPPORTED_TYPE_SUFFIXES
-
-        desired_types_processed: Set[str] = {
-            str(file_type) if str(file_type).startswith(".") else f".{str(file_type)}"
-            for file_type in desired_file_types
-        }
-        allowed_types: Set[str] = {
-            file_type for file_type in SUPPORTED_TYPE_SUFFIXES if file_type in desired_types_processed
-        }
-
-        return list(allowed_types)
-
     @staticmethod
     def _preprocess_paths(
         paths: List[Path],
         spinner: yaspin.Spinner = None,
         recursive: bool = False,
-        desired_file_types: Optional[List[str]] = None,
+        desired_file_types: List[str] | None = None,
```

Comment:
Nit: we could use sth like the following so that you can get rid of all the `None` checks

```python

class AlwaysContains:
    def __contains__(self, item):
        return True

fake_list = AlwaysContains()

print(42 in fake_list)  # True
print("hello" in fake_list)  # True
print(None in fake_list)  # True
```

Reply:
(it's in my opinion easy to miss the check somewhere)

Reply:
you mean replacing the native List objects with a class ? I would need to adjust it also within the typer Argument => CLI parsing, but might work.

Reply:
More replacing the `None` with an object behaving like a list (basically simulating the null pattern https://refactoring.guru/introduce-null-object